articles/node_lifecycle.md

```diff
@@ -0,0 +1,287 @@
+the node developer on how they provide the managed life cycle functionality,
+while also ensuring that any tools created for managing nodes can work with any
+compliant node.
+
+
+## Life cycle
+
+![The proposed node life cycle state machine](/img/node_lifecycle/life_cycle_sm.png "The proposed node life cycle state machine")
+
+The behaviour of each state is as defined below.
+
+### Created state
+
+This is the life cycle state the node is in immediately after being
+instantiated. The node may transition to the `inactive` state by being
+initialised.
+
+Implementation note: In object-oriented languages, the onCreated method can be
+mapped to the class constructor.
+
+### Inactive state
+
+This state represents a node that is not currently performing any processing.
+While in this state, the node will not receive any execution time to read
+topics, perform processing of data, respond to service requests, etc. Any data
+that arrives on topics will not be read. Any service requests will not be
+answered.
+
+[To discuss: All of the above, and additionally should data/service requests be
+queued for processing when the node is made active?]
+
+Entering the `inactive` state from the `created` state requires running the
+node's `onInitialised` method. The node uses this to set up any resources it
+must hold throughout its life (irrespective of if it is active or inactive).
+Such resources may include topic publications and subscriptions, memory that is
+held continuously, and initialising configuration parameters.
+
+Entering the `inactive` state from the `active` state requires running the
+node's `onDeactivated` method. See the `active` state description for more
+information.
+
+Entering the `inactive` state from the `fatal error` state requires running
+the node's `onReset` method. See the `fatal error` state description for more
+information.
+
+If an error occurs while in this state that cannot be handled by the node, then
+the transition to the `fatal error` state is taken.
+
+[Alternative: Do not transition, but return an "activation failed" error to
+whoever called the activate method/service.]
+
+The node may also exit this state by being destroyed, in which case the
+`destroy` transition is taken.
+
+Implementation note: Causing the transition to the destroyed state can often be
+mapped to deletion of the node instance.
+
+### Active state
+
+This is the main state of the node's life cycle. While in this state, the node
+performs any processing, responds to service requests, reads and processes
+data, produces output, etc.
+
+Entering the `active` state requires running the node's `onActivated` method.
+This is used to prepare the node for execution. This may include acquiring
+resources that are only held while the node is actually active, such as access
+to hardware. One common action that may be performed in `onActivated` is adding
+the node to an executor so that it receives execution time.
```

Comment:
I'm not sure what I was intending with that last line. I agree with you that executors should be owned by something else ("the system" will do for now) and nodes added to them.

One of my biggest complaints about the RTC specification is that it has a bizarre ownership relationship between components and execution contexts. It makes no sense and restricts how tools can be structured.